include/wx/persist/dataview.h

```diff
@@ -0,0 +1,125 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/persist/dataview.h
+// Purpose:     Persistence support for wxDataViewListCtrl
+// Author:      wxWidgets Team
+// Created:     2017-08-21
+// Copyright:   (c) 2017 wxWidgets.org
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PERSIST_DATAVIEW_H_
+#define _WX_PERSIST_DATAVIEW_H_
+
+#include "wx/persist/window.h"
+#include "wx/dataview.h"
+
+// ----------------------------------------------------------------------------
+// String constants used by wxPersistentDataViewListCtrl.
+// ----------------------------------------------------------------------------
+
+#define wxPERSIST_DVLC_KIND "DataViewList"
+
+#define wxPERSIST_DVLC_COL_PREFIX "Column"
+
+#define wxPERSIST_DVLC_WIDTH "Width"
+#define wxPERSIST_DVLC_SORT_KEY "SortKey"
+#define wxPERSIST_DVLC_SORT_ASC "SortAsc"
+
+// ----------------------------------------------------------------------------
+// wxPersistentDataViewListCtrl: Saves and restores user modified column widths
+// and single column sort order. 
+//
+// Future improvements could be to save and restore column order if the user
+// has changed it and multicolumn sorts.
+// ----------------------------------------------------------------------------
+
+class wxPersistentDataViewListCtrl : 
+                                public wxPersistentWindow<wxDataViewListCtrl>
+{
+public:
+    wxPersistentDataViewListCtrl(wxDataViewListCtrl* control)
+        : wxPersistentWindow<wxDataViewListCtrl>(control)
+    {
+    }
+
+    virtual void Save() const wxOVERRIDE
+    {
+        wxDataViewListCtrl* const control = Get();
+        
+        int sortColumn = -1;
+        
+        for ( unsigned int col = 0; col < control->GetColumnCount(); col++ )
+        {
+            // Create a prefix string to identify each column.
+            wxString columnPrefix;
+            columnPrefix.Printf("%s%d", wxPERSIST_DVLC_COL_PREFIX, col);
+
+            // Save the width of each column.
+            int width = control->GetColumn(col)->GetWidth();
+            SaveValue(columnPrefix + wxPERSIST_DVLC_WIDTH, width);
```

Comment:
The reason for doing it this way is so that it is possible to add other run-time  adjustable column attributes in future, e.g. hiding or column order. The keys for these would all be of the form ColNAttr, where N is the column index and Attr the run-time adjusted attribute.

For example the next line could be:
`SaveValue(columnPrefix + wxPERSIST_DVLC_HIDDEN, hidden)`